app/grandchallenge/algorithms/views.py

```diff
@@ -335,6 +336,39 @@ def get_context_data(self, *args, **kwargs):
         return context
 
 
+class AlgorithmImageMarkDesired(
+    LoginRequiredMixin, ObjectPermissionRequiredMixin, View
+):
+    permission_required = "algorithms.change_algorithmimage"
+    raise_exception = True
+
+    @cached_property
+    def algorithm(self) -> Algorithm:
+        return get_object_or_404(Algorithm, slug=self.kwargs["slug"])
+
+    @cached_property
+    def algorithm_image(self) -> Algorithm:
+        return get_object_or_404(AlgorithmImage, pk=self.kwargs["pk"])
+
+    def get_permission_object(self):
+        return self.algorithm_image
+
+    def get(self, *args, **kwargs):
+        self.algorithm_image.mark_desired_version()
```

Comment:
What happens here if the selected image is not in ECR or on SageMaker?

Reply:
It would still get marked as desired (the only we check in that method is whether `is_manifest_valid` is True. I'm only making the button available to users when an image is also on ECR and SageMaker, but I will add a check here as well.

Reply:
OK - but we will need it to work for images not on ECR and SageMaker too as they get cleaned up every evening.

Reply:
We shouldn't call validate docker image again, the validation takes a long time and doesn't need to be repeated if it has already been done.